contracts/potlock/src/potlock_storage.rs

```diff
@@ -0,0 +1,159 @@
+}
+
+impl<M: ManagedTypeApi> Project<M> {
+    pub fn new(
+        project_id: ProjectId,
+        potlock_id: PotlockId,
+        name: ManagedBuffer<M>,
+        description: ManagedBuffer<M>,
+        owner: ManagedAddress<M>,
+    ) -> Self {
+        Project {
+            project_id,
+            potlock_id,
+            name,
+            description,
+            owner,
+            status: Status::Inactive,
+        }
+    }
+}
+
+#[derive(TypeAbi, TopEncode, TopDecode, NestedEncode, NestedDecode, PartialEq, Debug)]
+pub struct UserDonations<M: ManagedTypeApi> {
+    pub user: ManagedAddress<M>,
+    pub donations: EsdtTokenPayment<M>,
+}
+
+#[multiversx_sc::module]
+pub trait PotlockStorage {
+    fn is_valid_potlock_id(&self, potlock_id: PotlockId) -> bool {
+        potlock_id >= 1 && potlock_id <= self.potlocks().len()
+    }
+
+    fn require_potlock_exists(&self, potlock_id: PotlockId) {
+        require!(
+            self.is_valid_potlock_id(potlock_id) && !self.potlocks().item_is_empty(potlock_id),
+            "Potlock doesn't exist!",
+        )
+    }
+
+    fn require_potlock_is_active(&self, potlock_id: PotlockId) {
+        let potlock = self.potlocks().get(potlock_id);
+        require!(potlock.status == Status::Active, "Pot is not active!",)
+    }
+
+    fn is_valid_project_id(&self, project_id: ProjectId) -> bool {
+        project_id >= 1 && project_id <= self.projects().len()
+    }
+
+    fn require_project_exists(&self, project_id: ProjectId) {
+        require!(
+            self.is_valid_project_id(project_id) && !self.projects().item_is_empty(project_id),
+            "Project doesn't exist!",
+        )
+    }
+
+    fn require_project_is_active(&self, project_id: ProjectId) {
+        let project = self.projects().get(project_id);
+        require!(project.status == Status::Active, "Project is not active!",)
+    }
+
+    fn require_correct_percentages(
+        &self,
+        project_percentages: MultiValueEncoded<ProjectPercentage>,
+    ) {
+        let mut total_perc: u64 = 0;
+        for pp in project_percentages {
+            let (_, perc) = pp.into_tuple();
+            total_perc += perc;
+        }
+        require!(
+            total_perc <= MAX_PERCENTAGE,
+            "Total percentages more than 100%"
+        );
+    }
+
+    #[view(getFeeTokenIdentifier)]
+    #[storage_mapper("feeTokenIdentifier")]
+    fn fee_token_identifier(&self) -> SingleValueMapper<TokenIdentifier>;
+
+    #[view(getFeeAmount)]
+    #[storage_mapper("feeAmount")]
+    fn fee_amount(&self) -> SingleValueMapper<BigUint>;
+
+    #[view(getPotlocks)]
+    #[storage_mapper("potlocks")]
+    fn potlocks(&self) -> VecMapper<Pot<Self::Api>>;
+
+    #[view(getProjects)]
+    #[storage_mapper("projects")]
+    fn projects(&self) -> VecMapper<Project<Self::Api>>;
+
+    #[view(feeAmountAcceptPots)]
+    #[storage_mapper("feeAmountAcceptedPots")]
+    fn fee_amount_accepted_pots(&self) -> SingleValueMapper<BigUint>;
```

Comment:
What's the usage of this storage? You only increment it in the acceptPot, so just to keep track of the total pot amounts over time? Do you need this in the SC stored?

Reply:
Removed.